### PR TITLE
apt: GetAppletInfo

### PR DESCRIPTION
**Pull request by mailwl**
this allow to run the game SoniPro (ソニプロ)

### CODE REVIEW

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-23T10:31:06Z: https://github.com/citra-emu/citra/compare/0d086616d1af239b8e41b246bea3c1ef85fc907f...mailwl:4b509c85e892e7903cf0e7bbca98d786de2a9f85

**Comment by JamePeng** [comment]
cmd_buf[2]-[3]	 u64, Title ID
I don't think the Title ID should be a constant.
http://3dbrew.org/wiki/Title_list

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-02-24T07:17:14Z: https://github.com/citra-emu/citra/compare/0d086616d1af239b8e41b246bea3c1ef85fc907f...mailwl:cacfd1afb90b5b9ef9c3639fe56d8b0f49efb8a1

**Comment by mailwl** [comment]
I add Title ID for swkbd/US only (used for SoniPro), don't remember, if this function used in other games

File: src/core/hle/service/apt/apt.cpp [hunk 1]
```diff
@@ -136,6 +140,33 @@ void GetAppletManInfo(Service::Interface* self) {
...
+            break;
+        default:
+            cmd_buff[2] = 0;
+            LOG_WARNING(Service_APT, "No title ID low for AppletID=0x%08X", app_id);
+            break;
+    }
+
+    cmd_buff[3] = TID_HI_APPLET;
+    cmd_buff[4] = 0; // NAND
+    cmd_buff[5] = true;
```
Comment (Subv) [inline src/core/hle/service/apt/apt.cpp]: Parameters 5 and 6 (Registered and Loaded) should be retrieved from the HLE::Applet class.

File: src/core/hle/service/apt/apt.cpp [hunk 2]
```diff
@@ -136,6 +140,33 @@ void GetAppletManInfo(Service::Interface* self) {
...
 
+
+void GetAppletInfo(Service::Interface* self) {
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+
+    AppletId app_id = static_cast<AppletId>(cmd_buff[1]);
+
+    cmd_buff[1] = RESULT_SUCCESS.raw; // No error
+
+    switch(app_id) {
```
Comment (Subv) [inline src/core/hle/service/apt/apt.cpp]: whitespace before the (

File: src/core/hle/service/apt/apt.cpp [hunk 3]
```diff
@@ -136,6 +140,33 @@ void GetAppletManInfo(Service::Interface* self) {
...
+            cmd_buff[2] = 0;
+            LOG_WARNING(Service_APT, "No title ID low for AppletID=0x%08X", app_id);
+            break;
+    }
+
+    cmd_buff[3] = TID_HI_APPLET;
+    cmd_buff[4] = 0; // NAND
+    cmd_buff[5] = true;
+    cmd_buff[6] = true;
+    cmd_buff[7] = static_cast<u32>(AppletAttr::SYS);
```
Comment (Subv) [inline src/core/hle/service/apt/apt.cpp]: Same with AppletAttr

**Comment by Subv** [comment]
Also, see #1266

**Comment by mailwl** [comment]
Ah, the function already implemented in #1266, i think, this PR not needed

**Comment by bunnei** [comment]
Closing in favor of #1266

**Comment by mailwl** [comment]
ah sorry! forgot to close by myself, because i want before add a comment for Subv about getting AppID from HLE::Applet too

**Pull request closed without merging**